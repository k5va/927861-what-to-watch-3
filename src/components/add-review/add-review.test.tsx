import * as React from "react";
import {AddReview} from "@components";
import {Provider} from "react-redux";
import {NameSpace} from "@store";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import {AppState, AuthorizationStatus} from "@consts";
import {Router} from "react-router-dom";
import {history} from "@routes";
import * as renderer from "react-test-renderer";

const movie = {
  id: `1`,
  title: `Die hard`,
  genre: `Drama`,
  year: 2019,
  duration: 123,
  backgroundImage: ``,
  cover: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  poster: `img/bg-the-grand-budapest-hotel.jpg`,
  src: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by
      concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
      Gustave prides himself on providing first-class service to the hotel's guests, including satisfying
      the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies
      mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in
      her murder.`,
  director: `Wes Andreson`,
  actors: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
  rating: {
    score: 8.9,
    count: 240
  },
  comments: [
    {
      id: `1`,
      text: `Bla Bla Bla`,
      author: `John Doe`,
      date: new Date(),
      rating: 9.0
    },
    {
      id: `2`,
      text: `Foo Foo Foo`,
      author: `Duffy Duck`,
      date: new Date(),
      rating: 2.0
    },
    {
      id: `3`,
      text: `Bark Bark Bark`,
      author: `Pluto`,
      date: new Date(),
      rating: 5.0
    },
  ]
};

const mockApi = {
  addComment() {
    return Promise.resolve([]);
  },
  checkAuthorizationStatus() {
    return Promise.resolve({
      "id": 1,
      "email": `Oliver.conner@gmail.com`,
      "name": `Oliver.conner`,
      "avatar_url": `img/1.png`
    });
  }
};
const mockStore = configureStore([thunk.withExtraArgument(mockApi)]);
const store = mockStore({
  [NameSpace.APP]: {
    appState: AppState.READY
  },
  [NameSpace.USER]: {
    authorizationStatus: AuthorizationStatus.NO_AUTH,
    user: null
  }
});

it(`AddReview should render correctly`, () => {
  const wrapper = renderer
    .create(
        <Router history={history}>
          <Provider store={store}>
            <AddReview movie={movie} />
          </Provider>
        </Router>,
        {
          createNodeMock: () => ({})
        }
    )
    .toJSON();
  expect(wrapper).toMatchSnapshot();
});
