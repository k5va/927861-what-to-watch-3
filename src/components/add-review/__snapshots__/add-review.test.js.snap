// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddReview should render correctly 1`] = `
<section
  className="movie-card movie-card--full"
>
  <div
    className="movie-card__header"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="Die hard"
        src=""
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <nav
        className="breadcrumbs"
      >
        <ul
          className="breadcrumbs__list"
        >
          <li
            className="breadcrumbs__item"
          >
            <a
              className="breadcrumbs__link"
              href="/films/c6133443-f839-4d75-aeca-4459c7677405"
              onClick={[Function]}
            >
              Die hard
            </a>
          </li>
          <li
            className="breadcrumbs__item"
          >
            <a
              className="breadcrumbs__link"
            >
              Add review
            </a>
          </li>
        </ul>
      </nav>
      <div
        className="user-block"
      >
        <a
          className="user-block__link"
          href="#"
          onClick={[Function]}
        >
          Sign in
        </a>
      </div>
    </header>
    <div
      className="movie-card__poster movie-card__poster--small"
    >
      <img
        alt="Die hard poster"
        height="327"
        src="img/bg-the-grand-budapest-hotel.jpg"
        width="218"
      />
    </div>
  </div>
  <div
    className="add-review"
  >
    <form
      action="#"
      className="add-review__form"
      onSubmit={[Function]}
    >
      <div
        className="rating"
      >
        <div
          className="rating__stars"
        >
          <input
            className="rating__input"
            id="star-1"
            name="rating"
            type="radio"
            value="1"
          />
          <label
            className="rating__label"
            htmlFor="star-1"
          >
            Rating 1
          </label>
          <input
            className="rating__input"
            id="star-2"
            name="rating"
            type="radio"
            value="2"
          />
          <label
            className="rating__label"
            htmlFor="star-2"
          >
            Rating 2
          </label>
          <input
            className="rating__input"
            defaultChecked={true}
            id="star-3"
            name="rating"
            type="radio"
            value="3"
          />
          <label
            className="rating__label"
            htmlFor="star-3"
          >
            Rating 3
          </label>
          <input
            className="rating__input"
            id="star-4"
            name="rating"
            type="radio"
            value="4"
          />
          <label
            className="rating__label"
            htmlFor="star-4"
          >
            Rating 4
          </label>
          <input
            className="rating__input"
            id="star-5"
            name="rating"
            type="radio"
            value="5"
          />
          <label
            className="rating__label"
            htmlFor="star-5"
          >
            Rating 5
          </label>
        </div>
      </div>
      <div
        className="add-review__text"
      >
        <textarea
          className="add-review__textarea"
          id="review-text"
          maxLength={400}
          minLength={50}
          name="review-text"
          onChange={[Function]}
          placeholder="Review text"
          required={true}
        />
        <div
          className="add-review__submit"
        >
          <button
            className="add-review__btn"
            disabled={true}
            type="submit"
          >
            Post
          </button>
        </div>
      </div>
    </form>
  </div>
</section>
`;
